src/inset.c

```diff
@@ -447,6 +447,9 @@ EXTERN_MSC int GMT_inset (void *V_API, int mode, void *args) {
 		GMT_Report (API, GMT_MSG_DEBUG, "inset: Removed inset file\n");
 		gmt_reload_history (API->GMT);
 		gmt_reload_settings (API->GMT);
+		/* Undo any shrink scaling memory */
+		API->inset_shrink_scale = 1.0;
```

Comment:
I'm afraid this precaution might not be enough as it only happens inside ``inset.c``
Because whatever is set in API carries to other modules calls in Julia/MEX this undoing should be done in ``end_session`` as well, I think.

Reply:
Hm, it is only set when gmt inset is called and shrinking is required, and it is guaranteed there will be a gmt inset end call to follow.  So it only lives as long as we are in an inset.  It wont hurt to add it to gmt end but not sure if it would make a difference.  Can you demonstrate the need?

Reply:
Imagine there is an error before the ``gmt inset end`` call? 
What I know is that there still some of these shits that manage to survive and there are cases where I'm forced to call a ``gmt("destroy")`` to workaround the unknown trouble.

Reply:
OK, I dont mind.  So you think gmt_end (which is called by GMT_Destroy_Session) is the place then?

Reply:
Yes, probably ``gmt_end``. It's the place where all this kind of stuff is undone.